src/test/scala/uk/gov/nationalarchives/dp/client/EntityClientTest.scala

```diff
@@ -359,423 +254,231 @@ abstract class EntityClientTest[F[_], S](preservicaPort: Int, secretsManagerPort
-    expectedError.left.map(err => {
+    expectedError.left.map { err =>
       err.getMessage should equal(
         s"Status code 500 calling http://localhost:$preservicaPort/api/accesstoken/login with method POST statusCode: 500, response: "
       )
-    })
+    }
+    verifyServerRequests(Nil)
   }
 
   "streamBitstreamContent" should "stream content to the provided function" in {
-    val url = s"http://localhost:$preservicaPort"
-    preservicaServer.stubFor(post(urlEqualTo(tokenUrl)).willReturn(ok(tokenResponse)))
-    preservicaServer.stubFor(get(urlEqualTo("/bitstream/1")).willReturn(ok("test")))
+    val endpoints = EntityClientEndpoints(preservicaServer)
+    val bitstreamUrl = endpoints.bitstreamOneContentUrl
+    endpoints.stubStreamBitstreamContent()
+    val bitstreamUrlToStreamFrom = endpoints.preservicaUrl + bitstreamUrl
+
     val client = testClient
 
-    val response = client.streamBitstreamContent[String](stream)(
-      s"$url/bitstream/1",
-      _ => cme.pure(s"Test return value")
-    )
+    val response =
+      client.streamBitstreamContent[String](stream)(bitstreamUrlToStreamFrom, _ => cme.pure(s"Test return value"))
     val responseValue = valueFromF(response)
     responseValue should equal("Test return value")
+    verifyServerRequests(List(bitstreamUrl))
   }
 
   "streamBitstreamContent" should "return an error if the server is unavailable" in {
-    val url = s"http://localhost:$preservicaPort"
-    val tokenUrl = "/api/accesstoken/login"
-    preservicaServer.stubFor(post(urlEqualTo(tokenUrl)).willReturn(serverError()))
+    val endpoints = EntityClientEndpoints(preservicaServer)
+    endpoints.serverErrorStub(MockPreservicaAPI.tokenUrl)
 
     val client = testClient
     val response =
-      client.streamBitstreamContent[Unit](stream)(s"$url/bitstream/1", _ => cme.unit)
+      client.streamBitstreamContent[Unit](stream)(s"https://unusedTestUrl.com", _ => cme.unit)
 
     val expectedError = valueFromF(cme.attempt(response))
 
     expectedError.isLeft should be(true)
-    expectedError.left.map(err => {
+    expectedError.left.map { err =>
       err.getMessage should equal(
         s"Status code 500 calling http://localhost:$preservicaPort/api/accesstoken/login with method POST statusCode: 500, response: "
       )
-    })
+    }
+    verifyServerRequests(Nil)
   }
 
   "metadataForEntity" should "return the correct metadata for a CO" in {
-    val entityId = UUID.randomUUID()
-    val entity = valueFromF(fromType("CO", entityId, Option("title"), Option("description"), deleted = false))
-    val urls = metadataForEntitySetUp(entity)
+    val entity = createEntity(ContentObject)
     val client = testClient
+    val endpoints = EntityClientEndpoints(preservicaServer, Some(entity))
+
+    val requestUrls = List(
+      endpoints.stubGetEntity(),
+      endpoints.stubGetIdentifierForEntity(),
+      endpoints.stubMetadataFragment(1, UUID.fromString("28d79967-24f8-4bfd-b420-a29be8694d66")),
```

Comment:
This uuid and the one below is used a few times. Is it worth sticking it in a variable?